modules/bitgo/test/v2/unit/tss/ecdsa.ts

```diff
@@ -0,0 +1,242 @@
+import {
+  readSignedMessage,
+  Ecdsa,
+  ECDSA,
+  ECDSAMethods,
+} from '@bitgo/sdk-core';
+import * as openpgp from 'openpgp';
+import * as should from 'should';
+
+type KeyShare = ECDSA.KeyShare;
+const encryptNShare = ECDSAMethods.encryptNShare;
+const createCombinedKey = ECDSAMethods.createCombinedKey;
+
+describe('test tss helper functions', function () {
+  let mpc: Ecdsa;
+
+  let userKeyShare: KeyShare;
+  let backupKeyShare: KeyShare;
+  let bitgoKeyShare: KeyShare;
+
+  let userKey;
+  let backupKey;
+  let bitgoKey;
+
+  let userGpgKeypair: { publicKey: string, privateKey: string };
+  let backupGpgKeypair: { publicKey: string, privateKey: string };
+  let bitgoGpgKeypair: { publicKey: string, privateKey: string };
+
+  let commonKeychain: string;
+
+  before(async function () {
+    mpc = new Ecdsa();
+    this.timeout(50000);
+
+    userKeyShare = await mpc.keyShare(1, 2, 3);
+    backupKeyShare = await mpc.keyShare(2, 2, 3);
+    bitgoKeyShare = await mpc.keyShare(3, 2, 3);
+
+    userKey = mpc.keyCombine(userKeyShare.pShare, [backupKeyShare.nShares[1], bitgoKeyShare.nShares[1]]);
+    backupKey = mpc.keyCombine(backupKeyShare.pShare, [userKeyShare.nShares[2], bitgoKeyShare.nShares[2]]);
+    bitgoKey = mpc.keyCombine(bitgoKeyShare.pShare, [backupKeyShare.nShares[3], userKeyShare.nShares[3]]);
+    
+    (userKey.xShare.y + userKey.xShare.chaincode).should.equal(backupKey.xShare.y + backupKey.xShare.chaincode);
+    (userKey.xShare.y + userKey.xShare.chaincode).should.equal(bitgoKey.xShare.y + bitgoKey.xShare.chaincode);
+    commonKeychain = userKey.xShare.y + userKey.xShare.chaincode;
+
+    userGpgKeypair = await openpgp.generateKey({
```

Comment:
Seems like `openpgp.generatekey` is an independent async operations for the three keys, you can actually speed up this test using `promise.all`having them executed some what parallel than serial